src/Temporalio/Worker/TemporalWorker.cs

```diff
@@ -92,7 +92,8 @@ public TemporalWorker(IWorkerClient client, TemporalWorkerOptions options)
                     OnTaskCompleted: options.OnTaskCompleted,
                     RuntimeMetricMeter: MetricMeter,
                     WorkerLevelFailureExceptionTypes: options.WorkflowFailureExceptionTypes,
-                    DisableEagerActivityExecution: options.DisableEagerActivityExecution));
+                    DisableEagerActivityExecution: options.DisableEagerActivityExecution,
+                    AssertValidActivity: (activityType) => activityWorker?.AssertValidActivity(activityType)));
```

Comment:
I wonder if we should fail if there is no activity worker? Technically it's not a valid local activity if there is no activity worker. Unsure how to best handle this, open to just about any approach.